src/wallet/spend.cpp

```diff
@@ -511,14 +513,28 @@ std::optional<SelectionResult> ChooseSelectionResult(const CWallet& wallet, cons
         results.push_back(*srd_result);
     }
 
-    if (results.size() == 0) {
+    if (results.empty()) {
         // No solution found
         return std::nullopt;
     }
 
+    std::vector<SelectionResult> eligible_results;
+    std::copy_if(results.begin(), results.end(), std::back_inserter(eligible_results), [coin_selection_params](const SelectionResult& result) {
+        const auto initWeight{coin_selection_params.m_subtract_fee_outputs ? 0 : coin_selection_params.tx_noinputs_size * WITNESS_SCALE_FACTOR};
```

Comment:
In dbeddc1f7f43c6df293133cd2b12a7187c0e321e "wallet: Check max tx weight in coin selector"

We still need to calculate the weight correctly even when subtracting the fee from outputs, so this should not be checking that.

```suggestion
        const auto initWeight{coin_selection_params.tx_noinputs_size * WITNESS_SCALE_FACTOR};
```